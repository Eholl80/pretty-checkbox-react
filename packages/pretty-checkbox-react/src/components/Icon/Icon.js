// @flow

import * as React from 'react';
import classNames from 'classnames';
import { warn, mergeRefs, isValidIconType, error, removeIndexFromArray } from '../../utils/utils';

import { type IconPropType } from '../../types';

export const Icon = React.memo<IconPropType>(({ icon, type }: IconPropType) => {
    if (!isValidIconType(type)) {
        error('icon type is invalid. Options are svg, icon, or image');
    }

    return icon
        ? React.cloneElement(
              icon,
              {
                  className: classNames(type, icon.props.className),
              },
              icon.props.children
          )
        : null;
});

Icon.displayName = 'PCR.Icon';
