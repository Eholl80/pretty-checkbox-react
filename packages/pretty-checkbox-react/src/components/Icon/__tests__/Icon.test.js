import React from 'react';
import { render } from '@testing-library/react';
import { Icon } from '../Icon';

import 'jest-dom/extend-expect';

describe('Icon tests', function() {
    it('should render an icon', function() {
        const { container, getByTestId } = render(
            <Icon type="icon" icon={<i data-testid="icon" className="mdi mdi-check" />} />
        );

        expect(getByTestId('icon')).toHaveClass('icon');
        expect(container).toMatchSnapshot();
    });

    it('should work with icon is null', function() {
        expect(() => {
            render(<Icon type="icon" icon={null} />);
        }).not.toThrow();
    });

    it('should show an error when icon type is invalid', function () {
        global.console.error = jest.fn();

        const { container } = render(
            <Icon type="dummy" icon={<i data-testid="icon" className="mdi mdi-check" />} />
        );

        expect(global.console.error).toHaveBeenCalled();
    });
});
