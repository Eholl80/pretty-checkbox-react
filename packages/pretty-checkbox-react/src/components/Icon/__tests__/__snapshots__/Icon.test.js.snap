// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon tests should render an icon 1`] = `
<div>
  <i
    class="icon mdi mdi-check"
    data-testid="icon"
  />
</div>
`;
